### ISSUE TITLE
[Bug]: SimpleKeywordTableIndex dependency on NLTK-Resource stopwords not found.

### ISSUE THREAD

**Issue by murthychetty** [run-llama/llama_index#11875]
### Bug Description

LookupError: 
**********************************************************************
  Resource stopwords not found.
  Please use the NLTK Downloader to obtain the resource:

  [31m>>> import nltk
  >>> nltk.download('stopwords')
  [0m
  For more information see: https://www.nltk.org/data.html

  Attempted to load corpora/stopwords

  Searched in:
    - '/root/nltk_data'
    - '/usr/local/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/local/lib/python3.11/site-packages/llama_index/core/_static/nltk_cache'
**********************************************************************

### Version

0.10.3

### Steps to Reproduce

Going through this link for using customRetrievers: https://github.com/run-llama/llama_index/blob/720959a926e14bd5e2bccd519927cccb6e6ceff7/docs/examples/query_engine/CustomRetrievers.ipynb#L17

`keyword_index = SimpleKeywordTableIndex(nodes, storage_context=storage_context)`

### Relevant Logs/Tracbacks

```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/nltk/corpus/util.py", line 84, in __load
    root = nltk.data.find(f"{self.subdir}/{zip_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nltk/data.py", line 583, in find
    raise LookupError(resource_not_found)
LookupError: 
**********************************************************************
  Resource stopwords not found.
  Please use the NLTK Downloader to obtain the resource:

  [31m>>> import nltk
  >>> nltk.download('stopwords')
  [0m
  For more information see: https://www.nltk.org/data.html

  Attempted to load corpora/stopwords

  Searched in:
    - '/root/nltk_data'
    - '/usr/local/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/local/lib/python3.11/site-packages/llama_index/core/_static/nltk_cache'
**********************************************************************


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/keyword_table/base.py", line 92, in __init__
    super().__init__(
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/base.py", line 91, in __init__
    index_struct = self.build_index_from_nodes(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/base.py", line 209, in build_index_from_nodes
    return self._build_index_from_nodes(nodes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/keyword_table/base.py", line 181, in _build_index_from_nodes
    self._add_nodes_to_index(index_struct, nodes, self._show_progress)
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/keyword_table/base.py", line 150, in _add_nodes_to_index
    keywords = self._extract_keywords(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/keyword_table/simple_base.py", line 32, in _extract_keywords
    return simple_extract_keywords(text, self.max_keywords_per_chunk)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/keyword_table/utils.py", line 17, in simple_extract_keywords
    tokens = [t for t in tokens if t not in globals_helper.stopwords]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/indices/keyword_table/utils.py", line 17, in <listcomp>
    tokens = [t for t in tokens if t not in globals_helper.stopwords]
                                            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/llama_index/core/utils.py", line 85, in stopwords
    self._stopwords = stopwords.words("english")
                      ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nltk/corpus/util.py", line 121, in __getattr__
    self.__load()
  File "/usr/local/lib/python3.11/site-packages/nltk/corpus/util.py", line 86, in __load
    raise e
  File "/usr/local/lib/python3.11/site-packages/nltk/corpus/util.py", line 81, in __load
    root = nltk.data.find(f"{self.subdir}/{self.__name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nltk/data.py", line 583, in find
    raise LookupError(resource_not_found)
LookupError: 
**********************************************************************
  Resource stopwords not found.
  Please use the NLTK Downloader to obtain the resource:

  [31m>>> import nltk
  >>> nltk.download('stopwords')
  [0m
  For more information see: https://www.nltk.org/data.html

  Attempted to load corpora/stopwords

  Searched in:
    - '/root/nltk_data'
    - '/usr/local/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/local/lib/python3.11/site-packages/llama_index/core/_static/nltk_cache'
**********************************************************************
```

**Comment by logan-markewich** [comment]
@murthychetty try running `pip install -U llama-index-core`

**Comment by logan-markewich** [comment]
This was fixed sometime after v0.10.3
